onnxruntime/core/session/inference_session.h

```diff
@@ -309,15 +371,28 @@ class InferenceSession {
   // if they need.
   std::shared_ptr<onnxruntime::Model> model_;
 
+  // used to hold the ModelProto parsed in an applicable ctor to be used while calling parameter-less Load()
+  std::unique_ptr<ONNX_NAMESPACE::ModelProto> model_proto_;
+
```

Comment:
Why are these members protected instead of private?

Reply:
model_ was protected. So I made the model_proto_ be protected as well for consistency. There are a few members that are protected. If you feel they can/should be made private, I'd rather tackle that in another PR.

Reply:
I made model_proto_ and the new methods/members that I added to private. I tried making model_ private, but it broke the build. So I guess some/all things in there are protected for a reason. I would like to defer experimenting with which ones can move to private until the cleanup exercise.